### DISCUSSION TITLE
Apply custom task_id to header in Celery chord

### DISCUSSION THREAD

**Discussion by P1roks** [celery/celery#8313] [Q&A]
I have a celery group which needs to generate a rapport of its execution when it is done and also allows for dynamic progress checking when it is still running, so I am using a chord

```
group_res = chord(tasks,body=gen_rapport()).apply_async().parent
group_res.save() # save so it is retrievable by using GroupResult.restore

return group_res.id
```
but the problem is that I need to get the group execution details in gen_rapport task and so I would like to know its ID before executing the full chord. Is there a way to do this without making a new class based on chord?

I tried using the options kwarg when creating tasks group but it doesn't work

```
tasks = group([some_task.s() for _ in range(100)] , options = ( {"task_id": "sample_id" } ))
```

**Comment by P1roks** [comment]
Ok I have found a way but turns out that celery is probably bugged
```
tasks = group(some_task.s() for _ in range(100), options={"task_id": "uuid"})

group_res = chord(tasks,body=gen_rapport()) # options is removed from header
group_res.tasks = tasks # this works

group_res.apply_async()

assert group_res.parent.id == "uuid" # true
```

**Comment by Nusnus** [comment]
Checkout Canvas Stamping, it might fit your needs:
https://docs.celeryq.dev/en/stable/userguide/canvas.html#stamping
